tests/deploy/ui/test_ui.py

```diff
@@ -0,0 +1,174 @@
+import socket
+import subprocess
+import sys
+import time
+
+import httpx
+import panel as pn
+import pytest
+from playwright.sync_api import Page, expect
+
+from ragna._utils import timeout_after
+from ragna.deploy import Config
+
+from ..utils import TestAssistant
+
+
+def get_available_port():
+    with socket.socket() as s:
+        s.bind(("", 0))
+        return s.getsockname()[1]
+
+
+@pytest.fixture(scope="session")
+def api_port():
+    return get_available_port()
+
+
+@pytest.fixture(scope="session")
+def ui_port():
+    return get_available_port()
```

Comment:
Why are we keeping the same port throughout the same session? Is this a requirement somehow?

Reply:
Ah, that was leftover from when everything was broken and I was trying to reduce the number of variables. Reverted.